src/meta/Intersection.ts

```diff
@@ -0,0 +1,167 @@
+import global from '@dojo/shim/global';
+import WeakMap from '@dojo/shim/WeakMap';
+import { Base } from './Base';
+
+interface IntersectionDetail {
+	entries: WeakMap<Element, IntersectionObserverEntry>;
+	intersectionObserver?: IntersectionObserver; // attached observer
+	options: IntersectionGetOptions;
+	root: string;
+	rootMargin: string | undefined;
```

Comment:
`rootMargin?: string;` over `rootMargin: string | undefined`?

Reply:
I did this because I can't assign undefined to an optional string when declaring an object of that type.